src/compareImage.js

```diff
@@ -1,53 +1,82 @@
-  if (distance < options.mismatchThreshold && diff.percent < options.mismatchThreshold) {
-    prefixedLogger.log('no mismatch found ✅');
-    return true;
-  }
+  const snapshotsDir = path.join(testRoot, '__image_snapshots__');
+  const snapshotPath = path.join(snapshotsDir, `${testConfig.testName}.snap.${testConfig.imageType}`);
 
-  if (options.saveDifferencifiedImage) {
+  let result = {};
+  const diffDir = path.join(snapshotsDir, '__differencified_output__');
+  const diffPath = path.join(diffDir, `${testConfig.testName}.differencified.${testConfig.imageType}`);
+  if (fs.existsSync(snapshotPath) && !testConfig.isUpdate) {
+    let snapshotImage;
     try {
-      const diffPath = `${options.testReports}/${testName}_differencified.png`;
-      diff.image.write(diffPath);
-      prefixedLogger.log(`saved the diff image to disk at ${diffPath}`);
+      snapshotImage = await Jimp.read(snapshotPath);
     } catch (err) {
-      prefixedLogger.error(`failed to save the diff image: ${err}`);
+      prefixedLogger.error(`failed to read reference image ${err}`);
+      return { matched: false };
+    }
+    let testImage;
+    try {
+      testImage = await Jimp.read(capturedImage);
+    } catch (err) {
+      prefixedLogger.error(`failed to read test image ${err}`);
+      return { matched: false };
+    }
+    prefixedLogger.log('comparing...');
+    const distance = Jimp.distance(snapshotImage, testImage);
+    const diff = Jimp.diff(snapshotImage, testImage, globalConfig.mismatchThreshold);
+    if (distance < globalConfig.mismatchThreshold && diff.percent < globalConfig.mismatchThreshold) {
+      prefixedLogger.log('no mismatch found ✅');
+      return { matched: true };
+    }
+    if (globalConfig.saveDifferencifiedImage) {
+      try {
+        if (!fs.existsSync(diffDir)) {
+          fs.mkdirSync(diffDir);
+        }
+        if (fs.existsSync(diffPath)) {
+          fs.unlinkSync(diffPath);
+        }
+        diff.image.write(diffPath);
+        prefixedLogger.log(`saved the diff image to disk at ${diffPath}`);
+      } catch (err) {
+        prefixedLogger.error(`failed to save the diff image: ${err}`);
+      }
     }
-  }
 
-  prefixedLogger.error(`mismatch found❗
-    Result:
-      distance: ${distance}
-      diff: ${diff.percent}
-      misMatchThreshold: ${options.mismatchThreshold}
-  `);
-  return false;
+    prefixedLogger.error(`mismatch found❗
+      Result:
+        distance: ${distance}
+        diff: ${diff.percent}
+        misMatchThreshold: ${globalConfig.mismatchThreshold}
+    `);
+    return { diffPath, matched: false };
+  }
+  prefixedLogger.log(`screenshot saved in -> ${snapshotPath}`);
+  if (fs.existsSync(diffPath)) {
+    fs.unlinkSync(diffPath);
+  }
+  if (!fs.existsSync(snapshotsDir)) {
+    fs.mkdirSync(snapshotsDir);
+  }
+  fs.writeFileSync(snapshotPath, capturedImage);
+  result = testConfig.isUpdate ? { updated: true } : { added: true };
```

Comment:
this `result` variable is declared at the top but then overwritten here.

Declare it here or remove it and just return the value

Reply:
Good point. That was left over from my refactoring. will fix it

Reply:
Done